### DISCUSSION TITLE
PhpSpreadsheet without Composer

### DISCUSSION THREAD

**Discussion by KK4HFJ** [PHPOffice/PhpSpreadsheet#1969] [Q&A] [answered]
Our environment does not use namespaces, composer, or autoloads. Looking for simple instructions on how to use PhpSpreadsheet without the corresponding luggage. All I need to do is import a table. Currently importing excel tables exported as csv files. Works very well, but trying to eliminate a step.

**Comment by MarkBaker** [comment, marked as answer]
Is the world regressing back to the 20th century?

You don't even mention what version of PHP you're running that prevents use of composer, namespacing or autoloading; but these are all very different things.

Official statement first:
> Composer is the only official and supported solution to use PhpSpreadsheet. Alternative custom solutions are possible, but I'd strongly advise against them. Not using composer will give you more work to set up, and potential issues to maintain.

It is possible to have composer installed locally on your development environment, run it to create its vendor folder, and then just push all of the generated files to the production server, with a single include to the `/vendor/autoload.php` from your main script. There is no need to actually run composer (or even have it installed) in your production environment. Without this, you will have problems with some of the dependencies.

If you can't use any autoloader for any reason, then you'll need to write a script that does all the includes for you (including dependencies). Note that by eliminating that "luggage", you're creating a lot more luggage in terms of having to load every file for every request, which is a big performance and memory overhead, and significant additional disk access. The whole purpose of an autoloader is to load only those files that are necessary for that request, eliminating much of the runtime luggage. The ability to autoload class files when required has been a standard part of PHP since version 5.3.0, nearly 12 years ago (June 2009).

As long as you're running a PHP version that works with namespaces (also present in the core language since PHP version 5.3.0), and an autoloader (whether that's the `/vendor/autoload.php`, a custom autoloader, or manual inclusion of all files) then I do not understand why namespacing should be an issue in any way. If it is a problem (clashes with other packages), then there are tools like PHP-Scoper (widely used by the authors of WP plug-ins to reduce the risk of potential clashes with other plug-ins) that can be used to try and avoid those issues. If namespacing is still a problem, then you'll need to manually edit every file to provide a full classname, and every reference to that file in the codebase to use that full classname.

If the reason that you can't use these is because you're running a version of PHP < 5.3.0, then you won't be able to run PHPSpreadsheet anyway.

If you do need unofficial support for this, then I'll happily charge my hourly rate: just give me a contact number, and I will fax you a full set of instructions.

**Reply by agung-laksono811** [reply]
this my number can u give me the intructions how to running phpSpreadsheet in php 5 or <7 i appraciate it 
6285156853901 whatsapp only

**Comment by Ngabia** [comment]
Check out this link on Stackoverflow
https://stackoverflow.com/questions/49943367/how-to-include-phpspreadsheet-library-without-composer#:~:text=Run%20PhpSpreadsheet%20locally%20without%20Composer&text=Composer%20is%20the%20only%20official,and%20potential%20issues%20to%20maintain.

**Reply by agung-laksono811** [reply]
ii was read this link but i still do not understand with the meanings and how to implementation in codes space
